### PR TITLE
feat: ET-1426: Added Azure support and updated readme

### PR DESCRIPTION
**Pull request by mattsalt123**
Updated README.md with instructions for getting API key from OpenAI or Azure and added additional config options for OpenAI and Azure. Also, modified api.ts to handle different API types (OpenAI or Azure) based on the config.

### CODE REVIEW

File: src/api.ts [hunk 1]
```diff
@@ -47,8 +48,26 @@ class OpenAi {
...
-    if (basePath) {
-      this.openAiApiConfiguration.basePath = basePath;
+    switch (apiType) {
+      case 'azure':
+        this.openAiApiConfiguration.baseOptions =  {
+          headers: {
+            "api-key": apiKey,
+          },
+          params: {
+            'api-version': '2023-07-01-preview',
```
Comment (covolution) [inline src/api.ts]: This needs to be a parameter in the config

File: src/commands/config.ts [hunk 2]
```diff
@@ -51,13 +52,13 @@ export const configValidators = {
...
-      'Must start with "sk-"'
+        value.startsWith('sk-') || value.match(/^[a-z0-9]{32}$/),
+      'Must start with "sk-" or a valid 32 character Azure OpenAI API key'
     );
     validateConfig(
       CONFIG_KEYS.OCO_OPENAI_API_KEY,
-      config[CONFIG_KEYS.OCO_OPENAI_BASE_PATH] || value.length === 51,
-      'Must be 51 characters long'
+      config[CONFIG_KEYS.OCO_OPENAI_BASE_PATH] || value.length === 51 || value.length === 32,
+      'Must be 51 or 32 characters long'
```
Comment (covolution) [inline src/commands/config.ts]: perhaps in brackets (openai) or (azure)

File: src/commands/config.ts [hunk 3]
```diff
@@ -120,16 +121,32 @@ export const configValidators = {
...
+
   [CONFIG_KEYS.OCO_MODEL](value: any) {
     validateConfig(
       CONFIG_KEYS.OCO_MODEL,
       [
         'gpt-3.5-turbo',
         'gpt-4',
         'gpt-3.5-turbo-16k',
+        'turbo-613',
+        'gpt4-613',
```
Comment (covolution) [inline src/commands/config.ts]: I think those are deployments rather than model names.

**Review by covolution — CHANGES_REQUESTED** [review]
Can you link to the original pull request in the readme? https://github.com/di-sukharev/opencommit/pull/167

File: src/api.ts [hunk 4]
```diff
@@ -47,8 +50,26 @@ class OpenAi {
...
+        if (basePath) {
+          this.openAiApiConfiguration.basePath = basePath + 'openai/deployments/' + DEPLOYMENT;
+        }
+        break;
+      case 'openai':
+      default:
+        if (basePath) {
+          this.openAiApiConfiguration.basePath = basePath;
+        }
+        break;
```
Comment (covolution) [inline src/api.ts]: not needed

File: src/commands/config.ts [hunk 5]
```diff
@@ -51,13 +54,13 @@ export const configValidators = {
     validateConfig(CONFIG_KEYS.OCO_OPENAI_API_KEY, value, 'Cannot be empty');
     validateConfig(
       CONFIG_KEYS.OCO_OPENAI_API_KEY,
-      value.startsWith('sk-'),
-      'Must start with "sk-"'
+        value.startsWith('sk-') || value.match(/^[a-z0-9]{32}$/),
```
Comment (covolution) [inline src/commands/config.ts]: what's the whitespace?

Comment (covolution) [reply src/commands/config.ts]: regex should be a constant

File: src/api.ts [hunk 6]
```diff
@@ -22,6 +22,8 @@ const config = getConfig();
 let maxTokens = config?.OCO_OPENAI_MAX_TOKENS;
 let basePath = config?.OCO_OPENAI_BASE_PATH;
 let apiKey = config?.OCO_OPENAI_API_KEY;
+let apiType = config?.OCO_OPENAI_API_TYPE || 'openai';
```
Comment (covolution) [inline src/api.ts]: Should lets be constants?

**Review by covolution — APPROVED** [review]

**Review by vestron286 — APPROVED** [review]
Good job!

**Pull request merged**